app/src/main/java/com/example/pictureupload/framework/db/PicSourceImpl.kt

```diff
@@ -0,0 +1,22 @@
+package com.example.pictureupload.framework.db
+
+import com.example.pictureupload.data.PicSource
+import com.example.pictureupload.domain.PicDetails
+
+class PicSourceImpl(private val picDao: PicDetailsDao) : PicSource{
+
+    override suspend fun insertPic(picDetails: PicDetails) {
+        val picEntity = PicDetailsEntity(picDetails.id, picDetails.path, picDetails.uploaded)
```

Comment:
Create a mapper from PicDetails to PicDetailsEntity

fun PicDetails.toEntity() : PicDetailsEntity {
}